// @remax/toutiao/cjs/api/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.hideTabBar = exports.hideTabBarRedDot = exports.removeTabBarBadge = exports.setTabBarBadge = exports.setTabBarItem = exports.setTabBarStyle = exports.showTabBar = exports.showTabBarRedDot = exports.onMemoryWarning = exports.checkFollowState = exports.followOfficialAccount = exports.interactionAnalysis = exports.createMapContext = exports.createLivePlayerContext = exports.getRecorderManager = exports.getBackgroundAudioManager = exports.createInnerAudioContext = exports.getMenuButtonBoundingClientRect = exports.getMenuButtonLayout = exports.setKeepScreenOn = exports.getConnectedWifi = exports.createRewardedVideoAd = exports.createIntersectionObserver = exports.createSelectorQuery = exports.getExtConfig = exports.getExtConfigSync = exports.vibrateLong = exports.vibrateShort = exports.scanCode = exports.onCompassChange = exports.startCompass = exports.stopCompass = exports.onAccelerometerChange = exports.startAccelerometer = exports.stopAccelerometer = exports.makePhoneCall = exports.getNetworkType = exports.onNetworkStatusChange = exports.getClipboardData = exports.setClipboardData = exports.chooseAddress = exports.getSetting = exports.openSetting = exports.authorize = exports.pay = exports.requestPayment = exports.reportAnalytics = exports.getUserInfo = exports.navigateBackMiniProgram = exports.navigateToMiniProgram = exports.checkSession = exports.login = exports.getFileInfo = exports.getFileSystemManager = exports.getSavedFileList = exports.openDocument = exports.removeSavedFile = exports.saveFile = exports.createCanvasContext = exports.navigateToVideoView = exports.hideShareMenu = exports.showShareMenu = exports.getLocation = exports.openLocation = exports.chooseVideo = exports.createVideoContext = exports.saveVideoToPhotosAlbum = exports.chooseImage = exports.getImageInfo = exports.previewImage = exports.saveImageToPhotosAlbum = exports.clearStorage = exports.clearStorageSync = exports.getStorage = exports.getStorageInfo = exports.getStorageInfoSync = exports.getStorageSync = exports.removeStorage = exports.removeStorageSync = exports.setStorage = exports.setStorageSync = exports.connectSocket = exports.uploadFile = exports.downloadFile = exports.request = exports.createAnimation = exports.pageScrollTo = exports.startPullDownRefresh = exports.stopPullDownRefresh = exports.setNavigationBarTitle = exports.hideLoading = exports.hideToast = exports.showActionSheet = exports.showLoading = exports.showModal = exports.showToast = exports.navigateBack = exports.navigateTo = exports.redirectTo = exports.reLaunch = exports.switchTab = exports.exitMiniProgram = exports.getLaunchOptionsSync = exports.getUpdateManager = exports.getSystemInfo = exports.getSystemInfoSync = exports.canIUse = exports.getAppStub = void 0;
var shared_1 = require("@remax/shared");
exports.getAppStub = getApp;
exports.canIUse = tt.canIUse;
exports.getSystemInfoSync = tt.getSystemInfoSync;
exports.getSystemInfo = shared_1.promisify(tt.getSystemInfo);
exports.getUpdateManager = tt.getUpdateManager;
exports.getLaunchOptionsSync = tt.getLaunchOptionsSync;
exports.exitMiniProgram = tt.exitMiniProgram;
exports.switchTab = shared_1.promisify(tt.switchTab);
exports.reLaunch = shared_1.promisify(tt.reLaunch);
exports.redirectTo = shared_1.promisify(tt.redirectTo);
exports.navigateTo = shared_1.promisify(tt.navigateTo);
exports.navigateBack = shared_1.promisify(tt.navigateBack);
exports.showToast = shared_1.promisify(tt.showToast);
exports.showModal = shared_1.promisify(tt.showModal);
exports.showLoading = shared_1.promisify(tt.showLoading);
exports.showActionSheet = shared_1.promisify(tt.showActionSheet);
exports.hideToast = shared_1.promisify(tt.hideToast);
exports.hideLoading = shared_1.promisify(tt.hideLoading);
exports.setNavigationBarTitle = tt.setNavigationBarTitle;
exports.stopPullDownRefresh = tt.stopPullDownRefresh;
exports.startPullDownRefresh = tt.startPullDownRefresh;
exports.pageScrollTo = shared_1.promisify(tt.pageScrollTo);
exports.createAnimation = tt.createAnimation;
exports.request = shared_1.promisify(tt.request);
exports.downloadFile = shared_1.promisify(tt.downloadFile);
exports.uploadFile = shared_1.promisify(tt.uploadFile);
exports.connectSocket = tt.connectSocket;
exports.setStorageSync = tt.setStorageSync;
exports.setStorage = shared_1.promisify(tt.setStorage);
exports.removeStorageSync = tt.removeStorageSync;
exports.removeStorage = shared_1.promisify(tt.removeStorage);
exports.getStorageSync = tt.getStorageSync;
exports.getStorageInfoSync = tt.getStorageInfoSync;
exports.getStorageInfo = shared_1.promisify(tt.getStorageInfo);
exports.getStorage = shared_1.promisify(tt.getStorage);
exports.clearStorageSync = tt.clearStorageSync;
exports.clearStorage = shared_1.promisify(tt.clearStorage);
exports.saveImageToPhotosAlbum = shared_1.promisify(tt.saveImageToPhotosAlbum);
exports.previewImage = shared_1.promisify(tt.previewImage);
exports.getImageInfo = shared_1.promisify(tt.getImageInfo);
exports.chooseImage = shared_1.promisify(tt.chooseImage);
exports.saveVideoToPhotosAlbum = shared_1.promisify(tt.saveVideoToPhotosAlbum);
exports.createVideoContext = tt.createVideoContext;
exports.chooseVideo = shared_1.promisify(tt.chooseVideo);
exports.openLocation = shared_1.promisify(tt.openLocation);
exports.getLocation = shared_1.promisify(tt.getLocation);
exports.showShareMenu = shared_1.promisify(tt.showShareMenu);
exports.hideShareMenu = shared_1.promisify(tt.hideShareMenu);
exports.navigateToVideoView = shared_1.promisify(tt.navigateToVideoView);
exports.createCanvasContext = tt.createCanvasContext;
exports.saveFile = shared_1.promisify(tt.saveFile);
exports.removeSavedFile = shared_1.promisify(tt.removeSavedFile);
exports.openDocument = shared_1.promisify(tt.openDocument);
exports.getSavedFileList = tt.getSavedFileList;
exports.getFileSystemManager = tt.getFileSystemManager;
exports.getFileInfo = tt.getFileInfo;
exports.login = shared_1.promisify(tt.login);
exports.checkSession = tt.checkSession;
exports.navigateToMiniProgram = tt.navigateToMiniProgram;
exports.navigateBackMiniProgram = shared_1.promisify(tt.navigateBackMiniProgram);
exports.getUserInfo = shared_1.promisify(tt.getUserInfo);
exports.reportAnalytics = tt.reportAnalytics;
exports.requestPayment = shared_1.promisify(tt.requestPayment);
exports.pay = shared_1.promisify(tt.pay);
exports.authorize = shared_1.promisify(tt.authorize);
exports.openSetting = shared_1.promisify(tt.openSetting);
exports.getSetting = shared_1.promisify(tt.getSetting);
exports.chooseAddress = shared_1.promisify(tt.chooseAddress);
exports.setClipboardData = shared_1.promisify(tt.setClipboardData);
exports.getClipboardData = tt.getClipboardData;
exports.onNetworkStatusChange = tt.onNetworkStatusChange;
exports.getNetworkType = shared_1.promisify(tt.getNetworkType);
exports.makePhoneCall = shared_1.promisify(tt.makePhoneCall);
exports.stopAccelerometer = shared_1.promisify(tt.stopAccelerometer);
exports.startAccelerometer = shared_1.promisify(tt.startAccelerometer);
exports.onAccelerometerChange = tt.onAccelerometerChange;
exports.stopCompass = shared_1.promisify(tt.stopCompass);
exports.startCompass = shared_1.promisify(tt.startCompass);
exports.onCompassChange = tt.onCompassChange;
exports.scanCode = shared_1.promisify(tt.scanCode);
exports.vibrateShort = shared_1.promisify(tt.vibrateShort);
exports.vibrateLong = shared_1.promisify(tt.vibrateLong);
exports.getExtConfigSync = tt.getExtConfigSync;
exports.getExtConfig = shared_1.promisify(tt.getExtConfig);
exports.createSelectorQuery = tt.createSelectorQuery;
exports.createIntersectionObserver = tt.createIntersectionObserver;
exports.createRewardedVideoAd = tt.createRewardedVideoAd;
exports.getConnectedWifi = shared_1.promisify(tt.getConnectedWifi);
exports.setKeepScreenOn = shared_1.promisify(tt.setKeepScreenOn);
exports.getMenuButtonLayout = tt.getMenuButtonLayout;
exports.getMenuButtonBoundingClientRect = tt.getMenuButtonBoundingClientRect;
exports.createInnerAudioContext = tt.createInnerAudioContext;
exports.getBackgroundAudioManager = tt.getBackgroundAudioManager;
exports.getRecorderManager = tt.getRecorderManager;
exports.createLivePlayerContext = tt.createLivePlayerContext;
exports.createMapContext = tt.createMapContext;
exports.interactionAnalysis = shared_1.promisify(tt.interactionAnalysis);
exports.followOfficialAccount = shared_1.promisify(tt.followOfficialAccount);
exports.checkFollowState = shared_1.promisify(tt.checkFollowState);
exports.onMemoryWarning = tt.onMemoryWarning;
exports.showTabBarRedDot = shared_1.promisify(tt.showTabBarRedDot);
exports.showTabBar = shared_1.promisify(tt.showTabBar);
exports.setTabBarStyle = shared_1.promisify(tt.setTabBarStyle);
exports.setTabBarItem = shared_1.promisify(tt.setTabBarItem);
exports.setTabBarBadge = shared_1.promisify(tt.setTabBarBadge);
exports.removeTabBarBadge = shared_1.promisify(tt.removeTabBarBadge);
exports.hideTabBarRedDot = shared_1.promisify(tt.hideTabBarRedDot);
exports.hideTabBar = shared_1.promisify(tt.hideTabBar);
