// @remax/cli/lib/build/utils/output.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.output = void 0;
const path = __importStar(require("path"));
const node_notifier_1 = __importDefault(require("node-notifier"));
const COLORS = {
    red: '\x1b[31m',
    green: '\x1b[32m',
    yellow: '\x1b[33m',
    blue: '\x1b[34m',
};
const RESET = '\x1b[0m';
exports.output = (content, color) => {
    const message = Array.isArray(content) ? content : [content];
    console.log(`${COLORS[color]}%s${RESET}`, ...message);
};
function log(type, message) {
    console[type](message);
}
function notice(message) {
    node_notifier_1.default.notify({
        title: 'Remax build error',
        message,
        icon: path.join(__dirname, '../../../error.png'),
    });
}
exports.default = {
    message: exports.output,
    error: (message) => log('error', message),
    warn: (message) => log('warn', message),
    notice,
};
