// @remax/runtime/cjs/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuntimeOptions = exports.unstable_batchedUpdates = exports.usePageContext = exports.createHostComponent = exports.createPortal = exports.createNativeComponent = exports.createPageConfig = exports.createAppConfig = exports.PluginDriver = exports.render = void 0;
var render_1 = require("./render");
Object.defineProperty(exports, "render", { enumerable: true, get: function () { return __importDefault(render_1).default; } });
var PluginDriver_1 = require("./PluginDriver");
Object.defineProperty(exports, "PluginDriver", { enumerable: true, get: function () { return __importDefault(PluginDriver_1).default; } });
var createAppConfig_1 = require("./createAppConfig");
Object.defineProperty(exports, "createAppConfig", { enumerable: true, get: function () { return __importDefault(createAppConfig_1).default; } });
var createPageConfig_1 = require("./createPageConfig");
Object.defineProperty(exports, "createPageConfig", { enumerable: true, get: function () { return __importDefault(createPageConfig_1).default; } });
var createNativeComponent_1 = require("./createNativeComponent");
Object.defineProperty(exports, "createNativeComponent", { enumerable: true, get: function () { return __importDefault(createNativeComponent_1).default; } });
var ReactPortal_1 = require("./ReactPortal");
Object.defineProperty(exports, "createPortal", { enumerable: true, get: function () { return ReactPortal_1.createPortal; } });
var RuntimeOptions = __importStar(require("./RuntimeOptions"));
exports.RuntimeOptions = RuntimeOptions;
var shared_1 = require("@remax/shared");
Object.defineProperty(exports, "createHostComponent", { enumerable: true, get: function () { return shared_1.createHostComponent; } });
var usePageContext_1 = require("./hooks/usePageContext");
Object.defineProperty(exports, "usePageContext", { enumerable: true, get: function () { return __importDefault(usePageContext_1).default; } });
__exportStar(require("./hooks"), exports);
var render_2 = require("./render");
exports.unstable_batchedUpdates = render_2.ReactReconcilerInst.batchedUpdates;
