// @remax/cli/lib/build/webpack/config.mini.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = __importDefault(require("fs"));
const path = __importStar(require("path"));
const webpack = __importStar(require("webpack"));
const webpack_chain_1 = __importDefault(require("webpack-chain"));
const mini_css_extract_plugin_1 = __importDefault(require("mini-css-extract-plugin"));
const copy_webpack_plugin_1 = __importDefault(require("copy-webpack-plugin"));
const webpack_virtual_modules_1 = __importDefault(require("webpack-virtual-modules"));
const webpack_bundle_analyzer_1 = require("webpack-bundle-analyzer");
const webpackbar_1 = __importDefault(require("webpackbar"));
const types_1 = require("@remax/types");
const shared_1 = require("@remax/shared");
const ejs_1 = __importDefault(require("ejs"));
const extensions_1 = require("../../extensions");
const getEntries_1 = __importDefault(require("../../getEntries"));
const TurboPages = __importStar(require("../utils/turboPages"));
const staticCompiler = __importStar(require("../babel/compiler/static"));
const app_1 = __importDefault(require("../babel/app"));
const page_1 = __importDefault(require("../babel/page"));
const pageEvent_1 = __importDefault(require("../babel/pageEvent"));
const appEvent_1 = __importDefault(require("../babel/appEvent"));
const fixRegeneratorRuntime_1 = __importDefault(require("../babel/fixRegeneratorRuntime"));
const componentManifest_1 = __importDefault(require("../babel/componentManifest"));
const RemaxPlugins = __importStar(require("./plugins"));
const css_1 = require("./config/css");
const baseConfig_1 = __importDefault(require("./baseConfig"));
function prepare(api, options, target) {
    const meta = api.getMeta();
    const turboPagesEnabled = options.turboPages && options.turboPages.length > 0;
    const stubModules = [types_1.Platform.ali, types_1.Platform.toutiao, types_1.Platform.wechat]
        .filter(name => target !== name)
        .reduce((acc, name) => [...acc, `${name}/esm/api`, `${name}/esm/hostComponents`], []);
    const publicPath = '/';
    return {
        meta,
        turboPagesEnabled,
        stubModules,
        publicPath,
    };
}
function resolveBabelConfig(options) {
    if (fs_1.default.existsSync(path.join(options.cwd, 'babel.config.js'))) {
        return path.join(options.cwd, 'babel.config.js');
    }
    return false;
}
function webpackConfig(api, options, target) {
    var _a;
    const config = new webpack_chain_1.default();
    baseConfig_1.default(config, options, target);
    const { meta, turboPagesEnabled, stubModules, publicPath } = prepare(api, options, target);
    const { app, pages } = getEntries_1.default(options, api);
    config.entry(app.name).add(app.filename);
    pages.forEach(p => {
        config.entry(p.name).add(p.filename);
    });
    config.devtool(false);
    config.resolve.extensions.merge(extensions_1.targetExtensions(options.target));
    config.output.filename('[name].js');
    config.output.globalObject(meta.global);
    config.output.publicPath(publicPath);
    config.optimization.runtimeChunk({ name: 'runtime' });
    config.optimization.splitChunks({
        cacheGroups: {
            remaxStyles: {
                name: 'remax-styles',
                test: new RegExp(`(.css|.less|.sass|.scss|.stylus|.styl|${api.meta.style})$`),
                chunks: 'initial',
                minChunks: 2,
                minSize: 0,
            },
            remaxVendors: {
                name: 'remax-vendors',
                test: extensions_1.moduleMatcher,
                chunks: 'initial',
                minChunks: 2,
                minSize: 0,
                priority: 2,
            },
        },
    });
    config.optimization.minimize((_a = options.minimize) !== null && _a !== void 0 ? _a : false);
    // turbo pages
    if (turboPagesEnabled) {
        // webpack chain 的配置顺序是反过来的
        config.module
            .rule('turbo-page')
            .pre()
            .use('turbo-page-postprocess')
            .loader('babel')
            .options({
            usePlugins: [staticCompiler.postProcess],
            reactPreset: false,
        })
            .end()
            .test(extensions_1.moduleMatcher)
            .use('turbo-page-render')
            .loader('babel')
            .options({
            usePlugins: [staticCompiler.render.bind(null, api)],
            reactPreset: false,
        })
            .end()
            .test(filename => {
            const { pages } = getEntries_1.default(options, api);
            return !!TurboPages.filter(pages, options).find(p => p.filename === shared_1.slash(filename));
        })
            .use('turbo-page-preprocess')
            .loader('babel')
            .options({
            usePlugins: [staticCompiler.preprocess],
            reactPreset: false,
        });
    }
    config.module
        .rule('js')
        .test(extensions_1.moduleMatcher)
        .use('babel')
        .loader('babel')
        .options({
        babelrc: false,
        configFile: resolveBabelConfig(options),
        usePlugins: [
            app_1.default(app.filename),
            page_1.default(options, api),
            appEvent_1.default(app.filename),
            pageEvent_1.default(options),
            componentManifest_1.default(api, config),
            fixRegeneratorRuntime_1.default(),
        ],
        reactPreset: true,
        api,
        compact: process.env.NODE_ENV === 'production',
    });
    config.module.rule('native-component').test(extensions_1.moduleMatcher).use('native-component').loader('nativeComponent').options({
        remaxOptions: options,
        api,
    });
    css_1.cssConfig(config, options, false);
    config.module.rule('stub').test(extensions_1.moduleMatcher).use('stub').loader('stub').options({
        modules: stubModules,
    });
    config.module
        .rule('image')
        .test(/\.(png|jpe?g|gif|svg)$/i)
        .use('file')
        .loader(require.resolve('file-loader'));
    config.module
        .rule('font')
        .test(/\.(ttf|eot|woff|woff2)$/i)
        .use('file')
        .loader(require.resolve('file-loader'));
    const runtimeOptionsTemplate = fs_1.default.readFileSync(path.resolve(__dirname, '../../../template/apply-runtime-options.js.ejs'), 'utf-8');
    const runtimeOptionsPath = shared_1.slash('node_modules/@remax/apply-runtime-options.js');
    config.entry(app.name).prepend('@remax/apply-runtime-options');
    const runtimeOptions = {
        pxToRpx: options.pxToRpx,
        debug: !!process.env.REMAX_DEBUG,
        platform: options.target,
        pluginFiles: api.getRuntimePluginFiles(),
        hostComponents: '[]',
        pageEvents: '{}',
        appEvents: '[]',
    };
    const virtualModules = new webpack_virtual_modules_1.default({
        [runtimeOptionsPath]: ejs_1.default.render(runtimeOptionsTemplate, runtimeOptions, { debug: false }),
    });
    config.plugin('webpack-virtual-modules').use(virtualModules);
    const publicDirPath = path.join(options.cwd, 'public');
    if (fs_1.default.existsSync(publicDirPath)) {
        config
            .plugin('webpack-copy-plugin')
            .use(copy_webpack_plugin_1.default, [[{ from: publicDirPath, to: path.join(options.cwd, options.output) }]]);
    }
    config.plugin('webpackbar').use(webpackbar_1.default, [{ name: target }]);
    config.plugin('mini-css-extract-plugin').use(mini_css_extract_plugin_1.default, [{ filename: `[name]${meta.style}` }]);
    config.plugin('remax-optimize-entries-plugin').use(RemaxPlugins.OptimizeEntries, [meta]);
    config.plugin('remax-native-files-plugin').use(RemaxPlugins.NativeFiles, [options, api]);
    config.plugin('remax-runtime-options-plugin').use(RemaxPlugins.RuntimeOptions, [options, api]);
    config.plugin('remax-coverage-ignore-plugin').use(RemaxPlugins.CoverageIgnore);
    if (options.analyze) {
        config.plugin('webpack-bundle-analyzer').use(webpack_bundle_analyzer_1.BundleAnalyzerPlugin);
    }
    const context = {
        config,
        webpack,
        addCSSRule: (ruleConfig) => {
            css_1.addCSSRule(config, options, false, ruleConfig);
        },
    };
    if (typeof options.configWebpack === 'function') {
        options.configWebpack(context);
    }
    api.configWebpack(context);
    const externals = config.get('externals');
    const runtimeOptionsExternal = {
        '/__remax_runtime_options__': `require('/__remax_runtime_options__')`,
    };
    if (Array.isArray(externals)) {
        config.set('externals', [...externals, runtimeOptionsExternal]);
    }
    else {
        config.set('externals', Object.assign(Object.assign({}, externals), runtimeOptionsExternal));
    }
    return config.toConfig();
}
exports.default = webpackConfig;
