// @remax/wechat/cjs/api/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.hideKeyboard = exports.getRealtimeLogManager = exports.hideHomeButton = exports.requestSubscribeMessage = exports.cloud = exports.createInterstitialAd = exports.createRewardedVideoAd = exports.createIntersectionObserver = exports.createSelectorQuery = exports.getExtConfig = exports.getExtConfigSync = exports.createWorker = exports.vibrateLong = exports.vibrateShort = exports.scanCode = exports.onMemoryWarning = exports.onGyroscopeChange = exports.startGyroscope = exports.stopGyroscope = exports.onDeviceMotionChange = exports.startDeviceMotionListening = exports.stopDeviceMotionListening = exports.onCompassChange = exports.startCompass = exports.stopCompass = exports.onAccelerometerChange = exports.startAccelerometer = exports.stopAccelerometer = exports.makePhoneCall = exports.getScreenBrightness = exports.onUserCaptureScreen = exports.setKeepScreenOn = exports.setScreenBrightness = exports.getNetworkType = exports.onNetworkStatusChange = exports.getHCEState = exports.onHCEMessage = exports.sendHCEMessage = exports.startHCE = exports.stopHCE = exports.getClipboardData = exports.setClipboardData = exports.getBatteryInfo = exports.getBatteryInfoSync = exports.closeBluetoothAdapter = exports.getBluetoothAdapterState = exports.getBluetoothDevices = exports.getConnectedBluetoothDevices = exports.onBluetoothAdapterStateChange = exports.onBluetoothDeviceFound = exports.openBluetoothAdapter = exports.startBluetoothDevicesDiscovery = exports.stopBluetoothDevicesDiscovery = exports.closeBLEConnection = exports.createBLEConnection = exports.getBLEDeviceCharacteristics = exports.getBLEDeviceServices = exports.notifyBLECharacteristicValueChange = exports.onBLECharacteristicValueChange = exports.onBLEConnectionStateChange = exports.readBLECharacteristicValue = exports.writeBLECharacteristicValue = exports.addPhoneContact = exports.connectWifi = exports.getConnectedWifi = exports.getWifiList = exports.onGetWifiList = exports.onWifiConnected = exports.setWifiList = exports.startWifi = exports.stopWifi = exports.getBeacons = exports.onBeaconServiceChange = exports.onBeaconUpdate = exports.startBeaconDiscovery = exports.stopBeaconDiscovery = exports.getWeRunData = exports.checkIsSoterEnrolledInDevice = exports.checkIsSupportSoterAuthentication = exports.startSoterAuthentication = exports.chooseInvoice = exports.chooseInvoiceTitle = exports.addCard = exports.openCard = exports.chooseAddress = exports.getSetting = exports.openSetting = exports.authorize = exports.requestPayment = exports.reportAnalytics = exports.reportMonitor = exports.getUserInfo = exports.getAccountInfoSync = exports.navigateBackMiniProgram = exports.navigateToMiniProgram = exports.checkSession = exports.login = exports.getFileInfo = exports.getFileSystemManager = exports.getSavedFileInfo = exports.getSavedFileList = exports.openDocument = exports.removeSavedFile = exports.saveFile = exports.canvasGetImageData = exports.canvasPutImageData = exports.canvasToTempFilePath = exports.createCanvasContext = exports.createOffscreenCanvas = exports.getShareInfo = exports.hideShareMenu = exports.showShareMenu = exports.updateShareMenu = exports.chooseLocation = exports.getLocation = exports.onLocationChange = exports.openLocation = exports.startLocationUpdate = exports.startLocationUpdateBackground = exports.stopLocationUpdate = exports.createCameraContext = exports.getRecorderManager = exports.startRecord = exports.stopRecord = exports.createLivePlayerContext = exports.createLivePusherContext = exports.getBackgroundAudioManager = exports.getBackgroundAudioPlayerState = exports.onBackgroundAudioPause = exports.onBackgroundAudioPlay = exports.onBackgroundAudioStop = exports.pauseBackgroundAudio = exports.playBackgroundAudio = exports.seekBackgroundAudio = exports.stopBackgroundAudio = exports.createAudioContext = exports.createInnerAudioContext = exports.getAvailableAudioSources = exports.pauseVoice = exports.playVoice = exports.setInnerAudioOption = exports.stopVoice = exports.chooseMedia = exports.chooseVideo = exports.createVideoContext = exports.saveVideoToPhotosAlbum = exports.chooseImage = exports.chooseMessageFile = exports.compressImage = exports.getImageInfo = exports.previewImage = exports.saveImageToPhotosAlbum = exports.createMapContext = exports.clearStorage = exports.clearStorageSync = exports.getStorage = exports.getStorageInfo = exports.getStorageInfoSync = exports.getStorageSync = exports.removeStorage = exports.removeStorageSync = exports.setStorage = exports.setStorageSync = exports.createUDPSocket = exports.offLocalServiceDiscoveryStop = exports.offLocalServiceFound = exports.offLocalServiceLost = exports.offLocalServiceResolveFail = exports.onLocalServiceDiscoveryStop = exports.onLocalServiceFound = exports.onLocalServiceLost = exports.onLocalServiceResolveFail = exports.startLocalServiceDiscovery = exports.stopLocalServiceDiscovery = exports.closeSocket = exports.connectSocket = exports.onSocketClose = exports.onSocketError = exports.onSocketMessage = exports.onSocketOpen = exports.sendSocketMessage = exports.uploadFile = exports.downloadFile = exports.request = exports.getSelectedTextRange = exports.offKeyboardHeightChange = exports.onKeyboardHeightChange = exports.offWindowResize = exports.onWindowResize = exports.getMenuButtonBoundingClientRect = exports.nextTick = exports.setTopBarText = exports.createAnimation = exports.pageScrollTo = exports.startPullDownRefresh = exports.stopPullDownRefresh = exports.loadFontFace = exports.hideTabBar = exports.hideTabBarRedDot = exports.removeTabBarBadge = exports.setTabBarBadge = exports.setTabBarItem = exports.setTabBarStyle = exports.showTabBar = exports.showTabBarRedDot = exports.setBackgroundColor = exports.setBackgroundTextStyle = exports.hideNavigationBarLoading = exports.setNavigationBarColor = exports.setNavigationBarTitle = exports.showNavigationBarLoading = exports.hideLoading = exports.hideToast = exports.showActionSheet = exports.showLoading = exports.showModal = exports.showToast = exports.navigateBack = exports.navigateTo = exports.redirectTo = exports.reLaunch = exports.switchTab = exports.getLogManager = exports.setEnableDebug = exports.offAppHide = exports.offAppShow = exports.offAudioInterruptionBegin = exports.offAudioInterruptionEnd = exports.offError = exports.offPageNotFound = exports.onAppHide = exports.onAppShow = exports.onAudioInterruptionEnd = exports.onError = exports.onPageNotFound = exports.getLaunchOptionsSync = exports.getUpdateManager = exports.getSystemInfo = exports.getSystemInfoSync = exports.arrayBufferToBase64 = exports.base64ToArrayBuffer = exports.canIUse = exports.getAppStub = void 0;
var shared_1 = require("@remax/shared");
exports.getAppStub = getApp;
exports.canIUse = wx.canIUse;
exports.base64ToArrayBuffer = wx.base64ToArrayBuffer;
exports.arrayBufferToBase64 = wx.arrayBufferToBase64;
exports.getSystemInfoSync = wx.getSystemInfoSync;
exports.getSystemInfo = shared_1.promisify(wx.getSystemInfo);
exports.getUpdateManager = wx.getUpdateManager;
exports.getLaunchOptionsSync = wx.getLaunchOptionsSync;
exports.onPageNotFound = wx.onPageNotFound;
exports.onError = wx.onError;
exports.onAudioInterruptionEnd = wx.onAudioInterruptionEnd;
exports.onAppShow = wx.onAppShow;
exports.onAppHide = wx.onAppHide;
exports.offPageNotFound = wx.offPageNotFound;
exports.offError = wx.offError;
exports.offAudioInterruptionEnd = wx.offAudioInterruptionEnd;
exports.offAudioInterruptionBegin = wx.offAudioInterruptionBegin;
exports.offAppShow = wx.offAppShow;
exports.offAppHide = wx.offAppHide;
exports.setEnableDebug = shared_1.promisify(wx.setEnableDebug);
exports.getLogManager = wx.getLogManager;
exports.switchTab = shared_1.promisify(wx.switchTab);
exports.reLaunch = shared_1.promisify(wx.reLaunch);
exports.redirectTo = shared_1.promisify(wx.redirectTo);
exports.navigateTo = shared_1.promisify(wx.navigateTo);
exports.navigateBack = shared_1.promisify(wx.navigateBack);
exports.showToast = shared_1.promisify(wx.showToast);
exports.showModal = shared_1.promisify(wx.showModal);
exports.showLoading = shared_1.promisify(wx.showLoading);
exports.showActionSheet = shared_1.promisify(wx.showActionSheet);
exports.hideToast = shared_1.promisify(wx.hideToast);
exports.hideLoading = shared_1.promisify(wx.hideLoading);
exports.showNavigationBarLoading = shared_1.promisify(wx.showNavigationBarLoading);
exports.setNavigationBarTitle = shared_1.promisify(wx.setNavigationBarTitle);
exports.setNavigationBarColor = shared_1.promisify(wx.setNavigationBarColor);
exports.hideNavigationBarLoading = shared_1.promisify(wx.hideNavigationBarLoading);
exports.setBackgroundTextStyle = shared_1.promisify(wx.setBackgroundTextStyle);
exports.setBackgroundColor = shared_1.promisify(wx.setBackgroundColor);
exports.showTabBarRedDot = shared_1.promisify(wx.showTabBarRedDot);
exports.showTabBar = shared_1.promisify(wx.showTabBar);
exports.setTabBarStyle = shared_1.promisify(wx.setTabBarStyle);
exports.setTabBarItem = shared_1.promisify(wx.setTabBarItem);
exports.setTabBarBadge = shared_1.promisify(wx.setTabBarBadge);
exports.removeTabBarBadge = shared_1.promisify(wx.removeTabBarBadge);
exports.hideTabBarRedDot = shared_1.promisify(wx.hideTabBarRedDot);
exports.hideTabBar = shared_1.promisify(wx.hideTabBar);
exports.loadFontFace = shared_1.promisify(wx.loadFontFace);
exports.stopPullDownRefresh = shared_1.promisify(wx.stopPullDownRefresh);
exports.startPullDownRefresh = shared_1.promisify(wx.startPullDownRefresh);
exports.pageScrollTo = shared_1.promisify(wx.pageScrollTo);
exports.createAnimation = wx.createAnimation;
exports.setTopBarText = shared_1.promisify(wx.setTopBarText);
exports.nextTick = wx.nextTick;
exports.getMenuButtonBoundingClientRect = wx.getMenuButtonBoundingClientRect;
exports.onWindowResize = wx.onWindowResize;
exports.offWindowResize = wx.offWindowResize;
exports.onKeyboardHeightChange = wx.onKeyboardHeightChange;
exports.offKeyboardHeightChange = wx.offKeyboardHeightChange;
exports.getSelectedTextRange = wx.getSelectedTextRange;
exports.request = shared_1.promisify(wx.request);
exports.downloadFile = shared_1.promisify(wx.downloadFile);
exports.uploadFile = shared_1.promisify(wx.uploadFile);
exports.sendSocketMessage = shared_1.promisify(wx.sendSocketMessage);
exports.onSocketOpen = wx.onSocketOpen;
exports.onSocketMessage = wx.onSocketMessage;
exports.onSocketError = wx.onSocketError;
exports.onSocketClose = wx.onSocketClose;
exports.connectSocket = wx.connectSocket;
exports.closeSocket = shared_1.promisify(wx.closeSocket);
exports.stopLocalServiceDiscovery = shared_1.promisify(wx.stopLocalServiceDiscovery);
exports.startLocalServiceDiscovery = shared_1.promisify(wx.startLocalServiceDiscovery);
exports.onLocalServiceResolveFail = wx.onLocalServiceResolveFail;
exports.onLocalServiceLost = wx.onLocalServiceLost;
exports.onLocalServiceFound = wx.onLocalServiceFound;
exports.onLocalServiceDiscoveryStop = wx.onLocalServiceDiscoveryStop;
exports.offLocalServiceResolveFail = wx.offLocalServiceResolveFail;
exports.offLocalServiceLost = wx.offLocalServiceLost;
exports.offLocalServiceFound = wx.offLocalServiceFound;
exports.offLocalServiceDiscoveryStop = wx.offLocalServiceDiscoveryStop;
exports.createUDPSocket = wx.createUDPSocket;
exports.setStorageSync = wx.setStorageSync;
exports.setStorage = shared_1.promisify(wx.setStorage);
exports.removeStorageSync = wx.removeStorageSync;
exports.removeStorage = shared_1.promisify(wx.removeStorage);
exports.getStorageSync = wx.getStorageSync;
exports.getStorageInfoSync = wx.getStorageInfoSync;
exports.getStorageInfo = shared_1.promisify(wx.getStorageInfo);
exports.getStorage = shared_1.promisify(wx.getStorage);
exports.clearStorageSync = wx.clearStorageSync;
exports.clearStorage = shared_1.promisify(wx.clearStorage);
exports.createMapContext = wx.createMapContext;
exports.saveImageToPhotosAlbum = shared_1.promisify(wx.saveImageToPhotosAlbum);
exports.previewImage = shared_1.promisify(wx.previewImage);
exports.getImageInfo = shared_1.promisify(wx.getImageInfo);
exports.compressImage = shared_1.promisify(wx.compressImage);
exports.chooseMessageFile = shared_1.promisify(wx.chooseMessageFile);
exports.chooseImage = shared_1.promisify(wx.chooseImage);
exports.saveVideoToPhotosAlbum = shared_1.promisify(wx.saveVideoToPhotosAlbum);
exports.createVideoContext = wx.createVideoContext;
exports.chooseVideo = shared_1.promisify(wx.chooseVideo);
exports.chooseMedia = shared_1.promisify(wx.chooseMedia);
exports.stopVoice = shared_1.promisify(wx.stopVoice);
exports.setInnerAudioOption = shared_1.promisify(wx.setInnerAudioOption);
exports.playVoice = shared_1.promisify(wx.playVoice);
exports.pauseVoice = shared_1.promisify(wx.pauseVoice);
exports.getAvailableAudioSources = shared_1.promisify(wx.getAvailableAudioSources);
exports.createInnerAudioContext = wx.createInnerAudioContext;
exports.createAudioContext = wx.createAudioContext;
exports.stopBackgroundAudio = shared_1.promisify(wx.stopBackgroundAudio);
exports.seekBackgroundAudio = shared_1.promisify(wx.seekBackgroundAudio);
exports.playBackgroundAudio = shared_1.promisify(wx.playBackgroundAudio);
exports.pauseBackgroundAudio = shared_1.promisify(wx.pauseBackgroundAudio);
exports.onBackgroundAudioStop = wx.onBackgroundAudioStop;
exports.onBackgroundAudioPlay = wx.onBackgroundAudioPlay;
exports.onBackgroundAudioPause = wx.onBackgroundAudioPause;
exports.getBackgroundAudioPlayerState = shared_1.promisify(wx.getBackgroundAudioPlayerState);
exports.getBackgroundAudioManager = wx.getBackgroundAudioManager;
exports.createLivePusherContext = wx.createLivePusherContext;
exports.createLivePlayerContext = wx.createLivePlayerContext;
exports.stopRecord = shared_1.promisify(wx.stopRecord);
exports.startRecord = shared_1.promisify(wx.startRecord);
exports.getRecorderManager = wx.getRecorderManager;
exports.createCameraContext = wx.createCameraContext;
exports.stopLocationUpdate = shared_1.promisify(wx.stopLocationUpdate);
exports.startLocationUpdateBackground = shared_1.promisify(wx.startLocationUpdateBackground);
exports.startLocationUpdate = shared_1.promisify(wx.startLocationUpdate);
exports.openLocation = shared_1.promisify(wx.openLocation);
exports.onLocationChange = wx.onLocationChange;
exports.getLocation = shared_1.promisify(wx.getLocation);
exports.chooseLocation = shared_1.promisify(wx.chooseLocation);
exports.updateShareMenu = shared_1.promisify(wx.updateShareMenu);
exports.showShareMenu = shared_1.promisify(wx.showShareMenu);
exports.hideShareMenu = shared_1.promisify(wx.hideShareMenu);
exports.getShareInfo = shared_1.promisify(wx.getShareInfo);
exports.createOffscreenCanvas = wx.createOffscreenCanvas;
exports.createCanvasContext = wx.createCanvasContext;
exports.canvasToTempFilePath = shared_1.promisify(wx.canvasToTempFilePath);
exports.canvasPutImageData = shared_1.promisify(wx.canvasPutImageData);
exports.canvasGetImageData = shared_1.promisify(wx.canvasGetImageData);
exports.saveFile = shared_1.promisify(wx.saveFile);
exports.removeSavedFile = shared_1.promisify(wx.removeSavedFile);
exports.openDocument = shared_1.promisify(wx.openDocument);
exports.getSavedFileList = shared_1.promisify(wx.getSavedFileList);
exports.getSavedFileInfo = shared_1.promisify(wx.getSavedFileInfo);
exports.getFileSystemManager = wx.getFileSystemManager;
exports.getFileInfo = shared_1.promisify(wx.getFileInfo);
exports.login = shared_1.promisify(wx.login);
exports.checkSession = shared_1.promisify(wx.checkSession);
exports.navigateToMiniProgram = shared_1.promisify(wx.navigateToMiniProgram);
exports.navigateBackMiniProgram = shared_1.promisify(wx.navigateBackMiniProgram);
exports.getAccountInfoSync = wx.getAccountInfoSync;
exports.getUserInfo = shared_1.promisify(wx.getUserInfo);
exports.reportMonitor = wx.reportMonitor;
exports.reportAnalytics = wx.reportAnalytics;
exports.requestPayment = shared_1.promisify(wx.requestPayment);
exports.authorize = shared_1.promisify(wx.authorize);
exports.openSetting = shared_1.promisify(wx.openSetting);
exports.getSetting = shared_1.promisify(wx.getSetting);
exports.chooseAddress = shared_1.promisify(wx.chooseAddress);
exports.openCard = shared_1.promisify(wx.openCard);
exports.addCard = shared_1.promisify(wx.addCard);
exports.chooseInvoiceTitle = shared_1.promisify(wx.chooseInvoiceTitle);
exports.chooseInvoice = shared_1.promisify(wx.chooseInvoice);
exports.startSoterAuthentication = shared_1.promisify(wx.startSoterAuthentication);
exports.checkIsSupportSoterAuthentication = shared_1.promisify(wx.checkIsSupportSoterAuthentication);
exports.checkIsSoterEnrolledInDevice = shared_1.promisify(wx.checkIsSoterEnrolledInDevice);
exports.getWeRunData = shared_1.promisify(wx.getWeRunData);
exports.stopBeaconDiscovery = shared_1.promisify(wx.stopBeaconDiscovery);
exports.startBeaconDiscovery = shared_1.promisify(wx.startBeaconDiscovery);
exports.onBeaconUpdate = wx.onBeaconUpdate;
exports.onBeaconServiceChange = wx.onBeaconServiceChange;
exports.getBeacons = shared_1.promisify(wx.getBeacons);
exports.stopWifi = shared_1.promisify(wx.stopWifi);
exports.startWifi = shared_1.promisify(wx.startWifi);
exports.setWifiList = shared_1.promisify(wx.setWifiList);
exports.onWifiConnected = wx.onWifiConnected;
exports.onGetWifiList = wx.onGetWifiList;
exports.getWifiList = shared_1.promisify(wx.getWifiList);
exports.getConnectedWifi = shared_1.promisify(wx.getConnectedWifi);
exports.connectWifi = shared_1.promisify(wx.connectWifi);
exports.addPhoneContact = shared_1.promisify(wx.addPhoneContact);
exports.writeBLECharacteristicValue = shared_1.promisify(wx.writeBLECharacteristicValue);
exports.readBLECharacteristicValue = shared_1.promisify(wx.readBLECharacteristicValue);
exports.onBLEConnectionStateChange = wx.onBLEConnectionStateChange;
exports.onBLECharacteristicValueChange = wx.onBLECharacteristicValueChange;
exports.notifyBLECharacteristicValueChange = shared_1.promisify(wx.notifyBLECharacteristicValueChange);
exports.getBLEDeviceServices = shared_1.promisify(wx.getBLEDeviceServices);
exports.getBLEDeviceCharacteristics = shared_1.promisify(wx.getBLEDeviceCharacteristics);
exports.createBLEConnection = shared_1.promisify(wx.createBLEConnection);
exports.closeBLEConnection = shared_1.promisify(wx.closeBLEConnection);
exports.stopBluetoothDevicesDiscovery = shared_1.promisify(wx.stopBluetoothDevicesDiscovery);
exports.startBluetoothDevicesDiscovery = shared_1.promisify(wx.startBluetoothDevicesDiscovery);
exports.openBluetoothAdapter = wx.openBluetoothAdapter;
exports.onBluetoothDeviceFound = wx.onBluetoothDeviceFound;
exports.onBluetoothAdapterStateChange = wx.onBluetoothAdapterStateChange;
exports.getConnectedBluetoothDevices = shared_1.promisify(wx.getConnectedBluetoothDevices);
exports.getBluetoothDevices = shared_1.promisify(wx.getBluetoothDevices);
exports.getBluetoothAdapterState = shared_1.promisify(wx.getBluetoothAdapterState);
exports.closeBluetoothAdapter = shared_1.promisify(wx.closeBluetoothAdapter);
exports.getBatteryInfoSync = wx.getBatteryInfoSync;
exports.getBatteryInfo = shared_1.promisify(wx.getBatteryInfo);
exports.setClipboardData = shared_1.promisify(wx.setClipboardData);
exports.getClipboardData = shared_1.promisify(wx.getClipboardData);
exports.stopHCE = shared_1.promisify(wx.stopHCE);
exports.startHCE = shared_1.promisify(wx.startHCE);
exports.sendHCEMessage = shared_1.promisify(wx.sendHCEMessage);
exports.onHCEMessage = wx.onHCEMessage;
exports.getHCEState = shared_1.promisify(wx.getHCEState);
exports.onNetworkStatusChange = wx.onNetworkStatusChange;
exports.getNetworkType = shared_1.promisify(wx.getNetworkType);
exports.setScreenBrightness = shared_1.promisify(wx.setScreenBrightness);
exports.setKeepScreenOn = shared_1.promisify(wx.setKeepScreenOn);
exports.onUserCaptureScreen = wx.onUserCaptureScreen;
exports.getScreenBrightness = shared_1.promisify(wx.getScreenBrightness);
exports.makePhoneCall = shared_1.promisify(wx.makePhoneCall);
exports.stopAccelerometer = shared_1.promisify(wx.stopAccelerometer);
exports.startAccelerometer = shared_1.promisify(wx.startAccelerometer);
exports.onAccelerometerChange = wx.onAccelerometerChange;
exports.stopCompass = shared_1.promisify(wx.stopCompass);
exports.startCompass = shared_1.promisify(wx.startCompass);
exports.onCompassChange = wx.onCompassChange;
exports.stopDeviceMotionListening = shared_1.promisify(wx.stopDeviceMotionListening);
exports.startDeviceMotionListening = shared_1.promisify(wx.startDeviceMotionListening);
exports.onDeviceMotionChange = wx.onDeviceMotionChange;
exports.stopGyroscope = shared_1.promisify(wx.stopGyroscope);
exports.startGyroscope = shared_1.promisify(wx.startGyroscope);
exports.onGyroscopeChange = wx.onGyroscopeChange;
exports.onMemoryWarning = wx.onMemoryWarning;
exports.scanCode = shared_1.promisify(wx.scanCode);
exports.vibrateShort = shared_1.promisify(wx.vibrateShort);
exports.vibrateLong = shared_1.promisify(wx.vibrateLong);
exports.createWorker = wx.createWorker;
exports.getExtConfigSync = wx.getExtConfigSync;
exports.getExtConfig = shared_1.promisify(wx.getExtConfig);
exports.createSelectorQuery = wx.createSelectorQuery;
exports.createIntersectionObserver = wx.createIntersectionObserver;
exports.createRewardedVideoAd = wx.createRewardedVideoAd;
exports.createInterstitialAd = wx.createInterstitialAd;
exports.cloud = wx.cloud;
exports.requestSubscribeMessage = shared_1.promisify(wx.requestSubscribeMessage);
exports.hideHomeButton = shared_1.promisify(wx.hideHomeButton);
exports.getRealtimeLogManager = wx.getRealtimeLogManager;
exports.hideKeyboard = shared_1.promisify(wx.hideKeyboard);
